increase packet version to 3

Logan007 opened the pull request:
This pull request increases the packet version to 3. **All edges and supernodes of your network** need to be updated in order to be able to communicate (again).

Fixes #375 

_______

~~Also, an SSE related version correction is included to allow correct compilation on older CPUs (it was meant to go in a separate pull request, but, well, git and me, you know…): Sometimes, `__SSE3__` is indicated even though the CPU does not have SSE3-support. That's why we need to check for `__SSE4_1`__ instead.~~ _will go into another pull request_

Commit:
fixed sse version

fengdaolong commented:
I think there may be changes in the network protocol in the future, such as the realization of automatic voting for super nodes, although no one has a clear plan for the development of v3.0.

Logan007 commented:
The latest changes broke a lot, so current _dev_ will not work with the version from a week ago anyway. So, we could change the version field now. I agree with you @fengdaolong, nobody knows when the next big bang will happen here. But at least from my side, I do not plan any compatibility breaking changes in the near future. Only some speed-ups… :smile:

Actually, I have drafted a few ideas for leveraging a true peer-to-peer character. On paper, it looked quite wild and in theory, it worked out quite well. Just coding would beyond my current abilities. Also, I did not overcome the flaw of **guaranteeing** to reach **all** nodes with a broadcast. But we can discuss, perhaps in the _Issue_ section? It will take me a while to write it all down. It could serve as a discussion starter.

fengdaolong commented:
I very much look forward to your writing your ideas for everyone to discuss, because your ideas are always refreshing and you are a very talented person.😄😄😄

Logan007 committed:
updated cc20.c (reverted to original)

...in an attempt to split this pull request in two making the SSE version change a separate one
